TimberWinR/Outputs/Redis.cs

```diff
@@ -23,26 +24,29 @@ public class RedisOutput : OutputSender
         private readonly int _port;
         private readonly int _timeout;
         private readonly object _locker = new object();
-        private readonly List<string> _jsonQueue;
-        // readonly Task _consumerTask;
+        private readonly List<string> _jsonQueue;       
         private readonly string[] _redisHosts;
         private int _redisHostIndex;
         private TimberWinR.Manager _manager;
         private readonly int _batchCount;
+        private int _currentBatchCount;
+        private readonly int _maxBatchCount;       
         private readonly int _interval;
         private readonly int _numThreads;
-
+        private readonly long[] _sampleQueueDepths;
+        private int _sampleCountIndex;
         private long _sentMessages;
         private long _errorCount;
         private long _redisDepth;
-
-        private int _maxQueueSize;
-        private bool _queueOverflowDiscardOldest;
+        private DateTime? _lastErrorTime;
+        private const int QUEUE_SAMPLE_SIZE = 30; // 30 samples over 2.5 minutes (default)
```

Comment:
It seems as if there is enough logic around the batch size counting and calculation to warrant a separate class. That might make it a little easier to reason about exactly what is happening.